scripts/psake-build.ps1

```diff
@@ -0,0 +1,190 @@
+        exec {
+            & $dotnetExe test $project
+        }
+    }
+}
+
+Task Publish.SesTest.Win64 -Depends Requires.DotNetExe, Restore.NuGetPackages {
+    exec {
+        & $dotnetExe publish $srcDir\sestest\sestest.csproj --self-contained --output $publishDir\sestest\win10-x64 --runtime win10-x64
+    }
+
+    exec {
+        compress-archive $publishDir\sestest\win10-x64\* $publishDir\sestest-win10-x64.zip
+    }
+}
+
+Task Publish.SesTest.Linux64 -Depends Requires.DotNetExe, Restore.NuGetPackages {
+    exec {
+        & $dotnetExe publish $srcDir\sestest\sestest.csproj --self-contained --output $publishDir\sestest\linux-x64 --runtime linux-x64
+    }
+
+    exec {
+        compress-archive $publishDir\sestest\linux-x64\* $publishDir\sestest-linux-x64.zip
+    }
+}
+
+## --------------------------------------------------------------------------------
+##   Configuration Targets
+## --------------------------------------------------------------------------------
+## Tasks used to configure the build
+
+Task Request.x64 {
+    $script:targetPlatform = "x64"
+}
+
+Task Request.x86 {
+    $script:targetPlatform = "x86"
+}
+
+Task Request.Debug {
+    $script:configuration = "Debug"
+}
+
+Task Request.Release {
+    $script:configuration = "Release"
+}
+
+## --------------------------------------------------------------------------------
+##   Prerequisite Targets
+## --------------------------------------------------------------------------------
+## Tasks used to ensure that prerequisites are available when needed. 
+
+Task Requires.Configuration {
+    if ($configuration -eq $null) {
+        $script:configuration = "Debug"
+    }
+
+    Write-Host "Build configuration is $configuration"
+}
+
+Task Requires.DotNetExe {
+    $script:dotnetExe = (get-command dotnet).Path
+
+    if ($dotnetExe -eq $null)
+    {
+        $script:dotnetExe = resolve-path $env:ProgramFiles\dotnet\dotnet.exe -ErrorAction SilentlyContinue
+    }
+    
+    if ($dotnetExe -eq $null)
+    {
+        throw "Failed to find dotnet.exe"
+    }
+
+    Write-Host "Dotnet executable: $dotnetExe"
+}
+
+Task Requires.MsBuild {
+    # prefer MSBuild from VS2017 if its there
+    $script:msbuildExe = resolve-path "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2017\*\MSBuild\*\Bin\MSBuild.exe" -ErrorAction SilentlyContinue
+
+    if ($msbuildExe -eq $null)
+    {
+        $script:msbuildExe = (get-command msbuild).Path
+    }
+
+    if ($msbuildExe -eq $null)
+    {
+        throw "Failed to find msbuild.exe"
+    }
+
+    Write-Host "MSBuild executable: $msbuildExe"
+}
+
+Task Requires.Platform {
+
+    if ($targetPlatform -eq $null) {
+        $script:targetPlatform = "Debug"
+    }
+
+    Write-Host "Target platform is $targetPlatform"
+}
+
+## --------------------------------------------------------------------------------
+##   Support Functions
+## --------------------------------------------------------------------------------
+
+formatTaskName { 
```

Comment:
Missing `function` here? 

Also stylistically I prefer `function foo($args) { ... }` rather than `param` inside it. Not sure about you 😁

Reply:
This is a **psake** thing - `formatTaskName` is an existing function that takes a block and stashes it away for **psake** to use; in other words, we're calling a function, not defining one.

Reply:
And fair enough about the style; will switch.

Reply:
Aha, special magic :)